src/borg/repository.py

```diff
@@ -1463,7 +1461,7 @@ def recover_segment(self, segment, filename):
                             if crc32(d[4:size]) & 0xffffffff != crc:
                                 d = d[1:]
                                 continue
-                            fd.write(d[:size])
+                            dst_fd.write(d[:size])
                             d = d[size:]
                 finally:
                     del d
```

Comment:
Hmm, guess I better invert the nesting of the context managers, so that the rename at exit of SaveFile does not happen before the mmap is gone and the src_fd is closed.